Source/JavaScriptCore/jit/ExecutableAllocator.cpp

```diff
@@ -1203,6 +1270,31 @@ void* ExecutableAllocator::getJumpIslandToConcurrently(void* from, void* newDest
     constexpr bool useMemcpy = false;
     return allocator->makeIsland(bitwise_cast<uintptr_t>(from), bitwise_cast<uintptr_t>(newDestination), concurrently, useMemcpy);
 }
+
+Vector<void*> ExecutableAllocator::allocateJumpIslandsToButDontPatch(void* from, void* newDestination)
+{
+    FixedVMPoolExecutableAllocator* allocator = g_jscConfig.fixedVMPoolExecutableAllocator;
+    if (!allocator)
+        RELEASE_ASSERT_NOT_REACHED();
+
+    Locker locker { allocator->getLock() };
+
+    // Concurrently is set to true here becasue we can't free these islands yet.
```

Comment:
How bad is it if we never free islands? The lifecycle seems very strange at the moment

Reply:
I am also wondering if that means it is possible for there to be a race here, where the island gets deallocated before being jitted. Of course, setting this to false does immediately cause a UAF in the same thread.

Reply:
So the only safe thing to do would be to never deallocate a jump island ever I think